ruby/ql/lib/codeql/ruby/frameworks/GraphQL.qll

```diff
@@ -289,6 +324,64 @@ private class GraphqlFieldArgumentDefinitionMethodCall extends GraphqlSchemaObje
 
   /** Gets the name of the argument (i.e. the first argument to this `argument` method call) */
   string getArgumentName() { result = this.getArgument(0).getConstantValue().getStringlikeValue() }
+
+  /** Gets the type of this argument */
+  GraphqlType getArgumentType() { result = this.getArgument(1) }
+
+  /**
+   * Gets the element type of this argument, if it is an array.
+   * For example if the argument type is `[String]`, this predicate yields `String`.
+   */
+  GraphqlType getArgumentElementType() {
+    result =
+      any(ArrayLiteral lit | lit = this.getArgument(1) and lit.getNumberOfElements() = 1)
+          .getElement(0)
+  }
+}
+
+private class GraphqlType extends ConstantAccess {
```

Comment:
I've not really dug into this, but is it helpful that we drop to the AST level for this rather than using `DataFlow::ConstantAccessNode`?

Reply:
Yeah, that seems like too big of a change for this PR. A follow-up would be good, but probably not super high priority.